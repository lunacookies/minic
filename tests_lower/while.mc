func main {
	var i = 0
	while i < 10
		set i = i + 1

	var x = 0
	var y = 0
	var a = 0
	while x < 100 {
		while y < 100 {
			set a = a + x * y
		}
	}
}
