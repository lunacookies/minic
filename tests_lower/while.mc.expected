func main
	var i i64
	var x i64
	var y i64
	var a i64
	{
		set i = 0
		while (i < 10)
			set i = (i + 1)
		set x = 0
		set y = 0
		set a = 0
		while (x < 100) {
			while (y < 100) {
				set a = (a + (x * y))
			}
		}
	}
